azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -450,9 +453,9 @@ private boolean progressGraph() throws IOException {
       Set<String> outNodeIds = node.getOutNodes();
       ExecutableFlowBase parentFlow = node.getParentFlow();
 
-      // If a job is seen as failed, then we set the parent flow to
+      // If a job is seen as failed or killed due to failing SLA, then we set the parent flow to
       // FAILED_FINISHING
-      if (node.getStatus() == Status.FAILED) {
+      if (node.getStatus() == Status.FAILED || (node.getStatus() == Status.KILLED && node.isKilledBySLA())) {
```

Comment:
You check two statuses here. One is `Killed`, the other is `isKilledBySLA`. Which one does it usually go firstly? What if we miss one round of `progressGraph`. What will happen?